pycalphad/core/calculate.py

```diff
@@ -452,44 +452,65 @@ def calculate(dbf, comps, phases, mode=None, output='GM', fake_points=False, bro
     # If we don't do this, sympy will get confused during substitution
-    statevar_dict = dict((v.StateVariable(key), unpack_condition(value)) for key, value in kwargs.items() if key in statevar_strings)
+    statevar_dict = dict((getattr(v, key), unpack_condition(value)) for key, value in kwargs.items() if key in statevar_strings)
     # Sort after default state variable check to fix gh-116
     statevar_dict = OrderedDict(sorted(statevar_dict.items(), key=lambda x: str(x[0])))
     str_statevar_dict = OrderedDict((str(key), unpack_condition(value)) for (key, value) in statevar_dict.items())
 
     # Build phase records if they weren't passed
     if phase_records is None:
         models = instantiate_models(dbf, comps, active_phases, model=model, parameters=parameters)
-        phase_records = build_phase_records(dbf, comps, active_phases, statevar_dict,
-                                            models=models, parameters=parameters,
-                                            output=output, callables=callables,
-                                            build_gradients=False, build_hessians=False,
-                                            verbose=kwargs.pop('verbose', False))
+        phase_records = PhaseRecordFactory(dbf, comps, statevar_dict, models, parameters=parameters)
     else:
         # phase_records were provided, instantiated models must also be provided by the caller
         models = model
         if not isinstance(models, Mapping):
             raise ValueError("A dictionary of instantiated models must be passed to `equilibrium` with the `model` argument if the `phase_records` argument is used.")
         active_phases_without_models = [name for name in active_phases if not isinstance(models.get(name), Model)]
-        active_phases_without_phase_records = [name for name in active_phases if not isinstance(phase_records.get(name), PhaseRecord)]
-        if len(active_phases_without_phase_records) > 0:
-            raise ValueError(f"phase_records must contain a PhaseRecord instance for every active phase. Missing PhaseRecord objects for {sorted(active_phases_without_phase_records)}")
         if len(active_phases_without_models) > 0:
             raise ValueError(f"model must contain a Model instance for every active phase. Missing Model objects for {sorted(active_phases_without_models)}")
 
     maximum_internal_dof = max(len(models[phase_name].site_fractions) for phase_name in active_phases)
 
+    phase_local_conditions = {key: unpack_condition(value)
+                                  for key, value in sorted(conditions.items(), key=lambda x: str(x[0]))
+                                  if isinstance(key, v.StateVariable) and hasattr(key, 'phase_name')}
+    str_phase_local_conditions = {str(k): v for k, v in phase_local_conditions.items()}
+    plc_shape = tuple(len(x) for x in phase_local_conditions.values())
+    # TODO: move state variable conditions into conditions dict
```

Comment:
Is this TODO talking about moving the phase local conditions into the regular conditions dict? something else? is it still relevant to keep in?

Reply:
I eventually want to get rid of `calculate(..., T=300, 1e5)` and only allow passing in a conditions dict. I don't think it's a blocker for this, and I expect I'll end up touching this code again to fully implement fixed-phase conditions.